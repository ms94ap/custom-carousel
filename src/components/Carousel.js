import React, { useState } from 'react'
import './../css/block.scss';
import { CSSTransition, SwitchTransition, TransitionGroup } from 'react-transition-group';

import blocks from './../icons.json'
import Card from './Card';
import ButtonControl from './ButtonControl';


const Carousel = () =>  {
  
	const [ currentBlock, setCurrentBlock] = useState(1)
	const [ isNext, setIsNext ] = useState(false)
  const randomImages = blocks.map(block => (block.images[Math.floor(Math.random() * block.images.length)]))
  const picturesPerSlide = 4
  const indexOfLastPicture = currentBlock * picturesPerSlide
  const indexOfFirstPicture = indexOfLastPicture - picturesPerSlide
  const currentPictures = randomImages.slice(indexOfFirstPicture, indexOfLastPicture)

  const imageList = currentPictures.map((image, index) =>{
		return (
			<Card key={index} image={image}/>
		)
	})

  const handlerPrev = () => {
		let index = currentBlock
		// setIsNext(false)
		setCurrentBlock( index - 1 )		
		
  }
  
  const handlerNext = () => {
		let index = currentBlock
		// setIsNext(true)
		setCurrentBlock( index + 1 )
	}
	
	const totalBlocks = Math.ceil(randomImages.length / picturesPerSlide)

	let index = currentBlock
	let	isnext = isNext

	console.log("index", index)
	return (
		<div className="container">
			<div className="carousel">
				<TransitionGroup>
					<CSSTransition
						key={index}
						// appear={true}
						classNames={{
							enter: handlerNext ? 'enter-next' : 'enter-prev',
            	enterActive: 'enter-active',
            	exit: 'leave',
            	// exitActive: handlerNext ? 'leave-active-next' : 'leave-active-prev'
						}}
						timeout={5000}
						
					>
						<div className="carousel_slide" key={index}>
							{imageList}
						</div>
						
					</CSSTransition>
					</TransitionGroup>
			</div>
			<ButtonControl
					 		disabled={ currentBlock === 1 }
							title="Previous"
							onClick={handlerPrev}
							direction="prev"
					 />
					<ButtonControl
						disabled = { currentBlock === totalBlocks }
						title="Next"
						onClick={handlerNext}
						direction="next"
					/>	
		</div>
	)
}

export default Carousel