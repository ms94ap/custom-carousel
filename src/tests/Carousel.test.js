import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import Carousel from '../components/Carousel'
// import ButtonControl from '../components/ButtonControl'
// import Card from '../components/Card'

Enzyme.configure({ adapter: new Adapter() })

describe("Carousel", () => {

	describe("rendering components", () =>{
		it('renders Carousel properly', () => {
			const component = shallow(<Carousel/>)
			expect(toJSON(component)).toMatchSnapshot()
		})
		// it('renders Card properly', () => {
		// 	const component = mount(<Card/>)
		// 	expect(component).toMatchSnapshot()
		// })
		// it('renders ButtonControl properly', () => {
		// 	const component = mount(<ButtonControl/>)
		// 	expect(component).toMatchSnapshot()
		// })
	})

})