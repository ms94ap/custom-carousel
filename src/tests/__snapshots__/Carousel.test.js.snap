// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel rendering components renders Carousel properly 1`] = `
<div
  className="container"
>
  <div
    className="carousel"
  >
    <CSSTransition
      classNames={
        Object {
          "enter": "enterAnimation",
          "enterActive": "enterActiveAnimation",
          "exit": "exitAnimation",
          "exitActive": "exitActiveAnimation",
        }
      }
      timeout={
        Object {
          "enter": 1000,
          "exit": 1000,
        }
      }
    >
      <div
        className="carousel_slide"
      >
        <Card
          image="https://images.unsplash.com/photo-1590710529625-089e0fce50dd?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=751&q=80"
          key="0"
        />
        <Card
          image="https://images.unsplash.com/photo-1551418988-c21e451467b7?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=667&q=80"
          key="1"
        />
        <Card
          image="https://images.unsplash.com/photo-1602002418323-be2d725a9a56?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80"
          key="2"
        />
        <Card
          image="https://images.unsplash.com/photo-1533653251153-deae1449c17d?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=619&q=80"
          key="3"
        />
      </div>
    </CSSTransition>
    <ButtonControl
      direction="next"
      onClick={[Function]}
      title="Next"
    />
  </div>
</div>
`;
